internal/controller/controller_test.go

```diff
@@ -0,0 +1,603 @@
+package controller
+
+import (
+	"github.com/snapp-incubator/route-to-contour-httpproxy/internal/config"
+	"github.com/snapp-incubator/route-to-contour-httpproxy/pkg/consts"
+	"github.com/snapp-incubator/route-to-contour-httpproxy/pkg/utils"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	routev1 "github.com/openshift/api/route/v1"
+	contourv1 "github.com/projectcontour/contour/apis/projectcontour/v1"
+	v12 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"context"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"fmt"
+)
+
+const (
+	APIVersionRoute   = "route.openshift.io/v1"
+	APIVersionService = "v1"
+	KindRoute         = "Route"
+	KindService       = "Service"
+	DefaultNamespace  = "default"
+	RouterName        = "default"
+	RouteTimeout      = "120s"
+
+	ServiceFooName             = "foo"
+	RouteFooName               = "foo"
+	RouteFooFQDN               = "test.apps.example.com"
+	RouteFooWildCardPolicyType = "None"
+
+	RateLimitRequests = 100
+	RouteIPWhiteList  = "1.1.1.1 8.8.8.8"
+)
+
+var (
+	err             error
+	ServiceWeight   int32 = 100
+	ServiceFooPorts       = []v12.ServicePort{{Name: "https", Port: 443}}
+	RouterTypeMeta        = v1.TypeMeta{Kind: KindRoute, APIVersion: APIVersionRoute}
+	ServiceTypeMeta       = v1.TypeMeta{Kind: KindService, APIVersion: APIVersionService}
+)
+
+var _ = Describe("Testing Route to HTTPProxy Controller", func() {
```

Comment:
Adding a test case for creating an httpproxy for multiple routes with the same host and different paths could be helpful.